go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -184,6 +199,27 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 	return j, nil
 }
 
+func addConfig(sparkConfig map[string]string, key string, value string) {
+
+	if strings.TrimSpace(value) != "true" {
+		return
+	}
+
+	matches := featureRegex.FindAllStringSubmatch(key, -1)
+	if len(matches) == 0 {
+		return
+	}
+	featureName := matches[0][len(matches[0])-1]
+
+	for _, feature := range GetSparkConfig().Features {
+		if feature.Name == featureName {
+			for k, v := range feature.SparkConfig {
+				sparkConfig[k] = v
```

Comment:
So if multiple features override the same spark configs, the last one wins... is that ok/intentional/undesired (so we can add validation)? whichever it's, I would add docs on the config to explain the behavior

Reply:
Ideally they should be unique so this can be a nested map here instead of list of features.  In future thou, I can see this being extended to check for values as well i.e. feature name + value can be unique so hence a struct makes better sense. 

For now, I think using the first matching config makes sense (instead of the last). Added that with comments.